tests/bdd/features/ha/core-agent/test_target_switchover.py

```diff
@@ -0,0 +1,309 @@
+def the_volume_republish_on_another_node_has_succeeded():
+    """the volume republish on another node has succeeded."""
+    try:
+        ApiClient.volumes_api().put_volume_target(
+            VOLUME_UUID, Protocol("nvmf"), node=NODE_NAME_2, republish="true"
+        )
+    except grpc.RpcError:
+        pytest.fail("Volume Republish Failed")
+
+
+@then("the newer target should have R/W access to the replicas")
+def the_newer_target_should_have_rw_access_to_the_replicas():
+    """the newer target should have R/W access to the replicas."""
+    newer_nexus = get_newer_nexus()
+    u = parse_uri(newer_nexus["device_uri"])
+
+    fio = Fio(
+        name="job", rw="randrw", traddr=u.hostname, subnqn=u.path[1:]
+    ).build_for_userspace()
+
+    try:
+        subprocess.run(fio, shell=True, check=True)
+    except subprocess.CalledProcessError:
+        assert False, "FIO is not expected to be errored out"
+
+
+@then("the older target should not have R/W access the replicas")
+def the_older_target_should_not_have_rw_access_the_replicas():
+    """the older target should not have R/W access the replicas."""
+    u = parse_uri(pytest.older_nexus_uri)
+
+    fio = Fio(
+        name="job", rw="randrw", traddr=u.hostname, subnqn=u.path[1:]
+    ).build_for_userspace()
+
+    try:
+        code = subprocess.run(fio, shell=True, check=True).returncode
+        assert code != 0, "Fio is not expected to execute successfully"
+    except subprocess.CalledProcessError:
+        remove_stale_files_on_error(u.hostname, u.path[1:])
+        assert True, "FIO is not expected to be errored out"
+
+
+# HELPER FUNCTIONS
+
+
+def get_newer_nexus():
+    volume = ApiClient.volumes_api().get_volume(VOLUME_UUID)
+    nexus = ApiClient.nexuses_api().get_nexus(volume["state"]["target"]["uuid"])
+    return nexus
+
+
+def kill_docker_container(name):
+    command = "docker kill {}".format(name)
+    try:
+        subprocess.run(command, shell=True, check=True)
+    except subprocess.CalledProcessError:
+        assert False, "Could not kill the docker container"
+
+
+def start_docker_container(name):
+    command = "docker start {}".format(name)
+    try:
+        subprocess.run(command, shell=True, check=True)
+    except subprocess.CalledProcessError:
+        assert False, "Could not kill the docker container"
+
+
+def simulate_network_failure(io_engine_ip, port):
+    command = RULE_APPEND.format(DEPLOYER_NETWORK_INTERFACE, io_engine_ip, port)
+    try:
+        subprocess.run(command, shell=True, check=True)
+    except subprocess.CalledProcessError:
+        assert False, "Error while adding IP table rule"
+
+
+def remove_network_failure(io_engine_ip, port):
+    command = RULE_REMOVE.format(DEPLOYER_NETWORK_INTERFACE, io_engine_ip, port)
+    try:
+        subprocess.run(command, shell=True, check=True)
+    except subprocess.CalledProcessError:
+        assert False, "Error while adding IP table rule"
+
+
+def cleanup_iptable_rules(io_engine_ip):
```

Comment:
wonder if we should have a script for this as well, in case process is terminated half-way through? So you can just run it manually.

Reply:
Although fixtures will cleanup after each test, but if the fixture itself terminated halfway we will have these entries. I will add a script as well. Thanks

Reply:
Done.